### ISSUE TITLE
[Sprint] Enable configure max epoch of scheduler

### ISSUE THREAD

**Issue by illian01** [Nota-NetsPresso/netspresso-trainer#349]
## Description

- Max epoch (or end epoch) of scheduler is automatically determined at `builder.py` by `num_epochs` of training configuration.

https://github.com/Nota-NetsPresso/netspresso-trainer/blob/09dbcde61cd0a2dc0162f99863cdb7463bf423fc/src/netspresso_trainer/schedulers/builder.py#L7-L18

- If we can change scheduler lifetime by configuration, this helps to optimize training settings.
  - e.g. In total 300 epochs, use cosine annealing until 285 epoch, and constant learning rate until training ends.

## Checklists

- [ ] I would create a corresponding branch for this issue from the designated (mostly `dev`) branch.
- [x] This issue only contains a preceding agreement between project owners.
